pkg/kv/kvserver/app_batch.go

```diff
@@ -88,6 +89,16 @@ func (b *appBatch) toCheckedCmd(
 		cmd.Cmd.LogicalOpLog = nil
 		cmd.Cmd.ClosedTimestamp = nil
 	} else {
+		// If the command was using the deprecated version of the MVCCStats proto,
+		// migrate it to the new version and clear out the field.
+		res := cmd.ReplicatedResult()
+		if deprecatedDelta := res.DeprecatedDelta; deprecatedDelta != nil {
```

Comment:
I think we can remove it already, look at hold this is!

https://github.com/cockroachdb/cockroach/pull/18828

The tricky thing with these migrations is typically that we need to have durably applied any log entries that could contain the old values (i.e. a non-nil deprecated delta in this case). But we have since had a few below-raft migrations which do exactly that (they flush the apply pipelines across the cluster).

So this is a free lunch to remove. Let's do that separately though. I'll add a TODO referencing newly filed https://github.com/cockroachdb/cockroach/issues/94039.